### PR TITLE
NEO-2312:Fixing badge position issue

### PR DESCRIPTION
**Pull request by enrique-prado**
[IconButton with Badge](https://deploy-preview-452--neo-react-library-storybook.netlify.app/?path=/story/components-iconbutton--with-badge)

**Before tagging the team for a review, I have done the following:**

- [x] run `yarn all` locally: ensures that all tests pass, formatting is done, types pass, and builds pass
- [x] reviewed my code changes
- [x] updated the link at the top of this PR (or remove it if not applicable)
- [x] added any important information to this PR (description, images, GIFs, ect.)
- [x] done an accessibility check on my work (tested with Chrome's `axe Dev Tools`, Mac's VoiceOver, etc.)
- [x] tagged `@avaya-dux/dux-design` if any visual changes have occurred
- [x] tagged `@avaya-dux/dux-devs`

When an IconButton is placed inside a table row, the table css messes up the position of the badge. This fixes the issue.

`@avaya-dux/dux-design`: Verify the position of the badge looks good.

`@avaya-dux/dux-devs`: Mostly css changes.

- shim file restores css position for the badge.

image

## Summary by CodeRabbit

- **New Features**
  - Added `aria-label` attributes for better accessibility in `IconButton` components.
  - Introduced a new `WithBadge` story for `IconButton` with additional props like `iconSize`, `variant`, `badge`, and `status`.

- **Style**
  - Added a new stylesheet `IconButton_shim.css` for enhanced button and badge styling.

- **Chores**
  - Updated the `@avaya/neo-react` package version from `1.1.31` to `1.1.32`.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
 	"name": "@avaya/neo-react",
-	"version": "1.1.31",
+	"version": "1.1.32",
```
Comment (joe-s-avaya) [inline package.json:3]: DO NOT BUMP VERSION

I had to merge some stuff that isn't ready to be pushed out. Please revert. I will be pushing an update by tomorrow EOD. (most likely today)

File: src/components/IconButton/IconButton_shim.css [hunk 2]
```diff
@@ -0,0 +1,8 @@
+.neo-btn.neo-icon-btn {
+	display: flex;
+}
+
+.neo-btn.neo-badge::after {
+	right: 7px;
+	top:7px;
```
Comment (joe-s-avaya) [inline src/components/IconButton/IconButton_shim.css:7]: you need to format

**Review by joe-s-avaya — COMMENTED** [review]
You need to add a test (a story) in the Table as you state that this is only an issue in the Table.

File: src/components/IconButton/IconButton_shim.css (hunk 2 above)
Comment (enrique-prado) [reply src/components/IconButton/IconButton_shim.css:7]: yarn all didn't catch this. fixed.

Comment (joe-s-avaya) [reply src/components/IconButton/IconButton_shim.css]: dang it, I was so certain that I'd setup the formatting command properly!

**Comment by matthewAvaya** [comment]
Position the badge so it vertically aligns with the top edge of the icon button as it currently sits lower than expected.

**Comment by enrique-prado** [comment]
Fixed vert alignment for badge as suggested by Matt.

**Pull request merged**